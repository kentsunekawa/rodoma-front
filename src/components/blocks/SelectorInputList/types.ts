export type Options = {
  value: number | string;
  label: string;
}[];

export interface SelectorData {
  selected: number | string;
  options?: Options;
}

export interface InputData {
  value: string;
  label?: string;
}

export interface ListData {
  selectorData: SelectorData;
  inputData: InputData;
}
