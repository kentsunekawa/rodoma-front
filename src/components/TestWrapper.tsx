import React from 'react';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from 'state/store';
import Theme from 'components/Theme';

interface Props {
  children?: React.ReactNode;
}

const Component: React.FC<Props> = (props: Props) => (
  <Provider store={store}>
    <Theme>
      <Router>{props.children}</Router>
    </Theme>
  </Provider>
);

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'queries'>) =>
  render(ui, { wrapper: Component, ...options });

export * from '@testing-library/react';

export { customRender as render };
