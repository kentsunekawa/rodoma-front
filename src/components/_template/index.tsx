import React from 'react';
import Styled from 'styled-components';
import * as styles from './styles';

// component root class name
const CLASSNAME = '';

// declare types
type StyleType = 'primary';

interface ComponentProps {
  className?: string;
  types?: StyleType[];
}

interface Props extends ComponentProps {}

// dom component
const Component: React.FC<Props> = (props) => (
  <div className={`${CLASSNAME} ${props.className}`}></div>
);

// styled component
const StyeldComponent = Styled(Component)`
  ${styles.base}
  // extended styles
  ${({ types }) => types && types.map((type) => styles[type])}}
`;

// container component
const Container: React.FC<ComponentProps> = (componentProps) => {
  const props = {};

  return <StyeldComponent {...componentProps} {...props}></StyeldComponent>;
};
export default Container;
