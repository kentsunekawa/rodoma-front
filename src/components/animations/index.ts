import gsap from "gsap";

export const doorHide = (root: HTMLDivElement, logo: HTMLDivElement) => {
  return new Promise((resolve) => {
    gsap.to(logo, {
      y: "-200px",
      duration: 1,
      ease: "Power3.easeOut",
    });
    gsap.to(root, {
      opacity: 0,
      duration: 0.5,
      ease: "linear",
      onComplete: () => {
        gsap.set(root, {
          display: "none",
        });
        resolve(null);
      },
    });
  });
};

export const toggleLoading = (root: HTMLDivElement, dir: boolean = true) => {
  if (dir) {
    return new Promise((resolve) => {
      gsap.killTweensOf(root);
      gsap.set(root, {
        display: "block",
        opacity: "0",
      });
      gsap.to(root, {
        opacity: 1,
        duration: 0.2,
        ease: "linear",
        onComplete: () => {
          resolve(null);
        },
      });
    });
  } else {
    return new Promise((resolve) => {
      gsap.killTweensOf(root);
      gsap.to(root, {
        opacity: 0,
        duration: 0.2,
        ease: "linear",
        onComplete: () => {
          gsap.set(root, {
            display: "none",
          });
          resolve(null);
        },
      });
    });
  }
};

export const toggleMenu = (
  root: HTMLDivElement,
  panel: HTMLDivElement,
  overlay: HTMLDivElement,
  dir: boolean = true
) => {
  if (dir) {
    gsap.set(root, {
      display: "block",
    });
    gsap.set(overlay, {
      opacity: 0,
    });
    gsap.set(panel, {
      x: "100%",
    });
    gsap.to(overlay, {
      opacity: 1,
      duration: 0.3,
      ease: "Power1.easeInOut",
    });
    gsap.to(panel, {
      x: "0",
      duration: 0.3,
    });
  } else {
    gsap.to(overlay, {
      opacity: 0,
      duration: 0.3,
    });
    gsap.to(panel, {
      x: "100%",
      duration: 0.3,
      ease: "Power1.easeInOut",
      onComplete: () => {
        root.style.display = "none";
      },
    });
    gsap.set(root, {
      display: "none",
      delay: 0.3,
    });
  }
};

export const toggleAccordion = (
  child: HTMLDivElement,
  inner: HTMLDivElement,
  dir: boolean = true
) => {
  const ease = "Power2.easeInOut";
  const duration = 0.3;

  if (dir) {
    gsap.set(child, {
      height: 0,
    });
    gsap.to(child, {
      height: inner.clientHeight,
      duration,
      ease,
    });
  } else {
    gsap.to(child, {
      height: 0,
      duration,
      ease,
    });
  }
};

export const toggleSearchPanel = (
  bg: HTMLDivElement,
  inner: HTMLDivElement,
  dir: boolean = true
) => {
  const ease = "Power2.easeOut";
  const duration = 0.3;

  if (dir) {
    gsap.set(bg, {
      display: "block",
      opacity: 0,
    });
    gsap.set(inner, {
      opacity: 0,
      y: "50px",
    });
    gsap.to(bg, {
      opacity: 1,
      duration,
      ease: "Power0.easeNone",
    });
    gsap.to(inner, {
      y: 0,
      opacity: 1,
      duration,
      ease,
      delay: 0.1,
    });
  } else {
    gsap.to(inner, {
      y: "50px",
      opacity: 0,
      duration,
      ease,
    });
    gsap.to(bg, {
      opacity: 0,
      duration,
      ease: "Power0.easeNone",
      // delay: .2,
      onComplete: () => {
        bg.style.display = "none";
      },
    });
  }
};

export const toggleModal = (
  root: HTMLDivElement,
  mask: HTMLDivElement,
  overlay: HTMLDivElement,
  dir: boolean = true
) => {
  const duration = 0.3;
  const ease = "Power2.easeOut";

  return new Promise((resolve) => {
    if (dir) {
      gsap.set(root, {
        display: "block",
      });
      gsap.set(overlay, {
        opacity: 0,
      });
      gsap.set(mask, {
        y: "50px",
        opacity: 0,
      });
      gsap.to(overlay, {
        opacity: 1,
        duration,
        ease: "Power0.easeNone",
      });
      gsap.to(mask, {
        y: 0,
        opacity: 1,
        duration,
        ease,
        onComplete() {
          resolve(null);
        },
      });
    } else {
      gsap.to(overlay, {
        opacity: 0,
        duration,
        ease: "Power0.easeNone",
      });
      gsap.to(mask, {
        y: "50px",
        opacity: 0,
        duration,
        ease,
        onComplete() {
          resolve(null);
        },
      });
    }
  });
};

export const toggleMessage = (
  root: HTMLDivElement,
  counter: HTMLSpanElement,
  isShow: boolean = true,
  count?: number
) => {
  return new Promise((resolve) => {
    if (isShow) {
      gsap.killTweensOf(counter);
      gsap.set(root, {
        display: "block",
        opacity: 0,
      });
      gsap.set(counter, {
        width: 0,
      });
      gsap.to(counter, {
        width: "100%",
        duration: count,
        ease: "linear",
      });
      gsap.to(root, {
        opacity: 1,
        duration: 0.3,
        ease: "Power0.easeNone",
        onComplete: () => {
          resolve(null);
        },
      });
    } else {
      gsap.to(root, {
        opacity: 0,
        duration: 0.3,
        ease: "Power0.easeNone",
        onComplete: () => {
          gsap.set(root, {
            display: "none",
          });
          resolve(null);
        },
      });
    }
  });
};

export const toggleCard = (
  root: HTMLDivElement,
  panel: HTMLDivElement,
  overlay: HTMLDivElement,
  dir: boolean = true
) => {
  const duration = 0.4;

  return new Promise((resolve) => {
    if (dir) {
      gsap.set(root, {
        display: "block",
      });
      gsap.set(overlay, {
        opacity: 0,
      });
      gsap.set(panel, {
        bottom: "-100%",
      });
      gsap.to(overlay, {
        opacity: 1,
        duration,
        ease: "linear",
      });
      gsap.to(panel, {
        bottom: 0,
        duration,
        ease: "Power3.easeOut",
        onComplete: () => {
          resolve(null);
        },
      });
    } else {
      gsap.to(panel, {
        bottom: "-100%",
        duration,
        ease: "Power3.easeIn",
      });
      gsap.to(overlay, {
        opacity: 0,
        duration,
        ease: "linear",
        onComplete: () => {
          gsap.set(root, {
            display: "none",
          });
          resolve(null);
        },
      });
    }
  });
};
