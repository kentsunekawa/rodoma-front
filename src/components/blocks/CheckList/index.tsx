import React from 'react';
import Styled from 'styled-components';
import * as styles from './styles';
import CheckBox, { StyleType } from 'components/blocks/CheckList/CheckBox';

// component root class name
const CLASSNAME = 'CheckList';

interface CheckListValues {
  value: string | number;
  label: string;
};

interface ComponentProps {
  name: string;
  values: CheckListValues[];
  selected: (number | string)[],
  boxTypes?: StyleType[];
  className?: string;
  onChange: (values: (string | number)[]) => void;
}

interface Props extends ComponentProps {
  checkBoxes: React.ReactNode;
}

// dom component
const Component: React.FC<Props> = props => (
  <div className={`${CLASSNAME} ${props.className}`}>
    <ul className='list'>
      {props.checkBoxes}
    </ul>
  </div>
);

// styled component
const StyeldComponent = Styled(Component)`
  ${styles.base}
`;

// container component
const Container: React.FC<ComponentProps> = componentPrps => {

  const { name, values, boxTypes, selected, onChange } = componentPrps;

  const onChildChange = (value: string | number) => {
    let newSelected = selected.slice();
    if(newSelected.includes(value)){
      newSelected.splice(newSelected.indexOf(value), 1);
    } else {
      newSelected.push(value);
    } 
    onChange(newSelected); 
  }

  const checkBoxes = values.map((value, i) => {
    return <li className="item" key={i}>
      <CheckBox
        isChecked={selected.includes(value.value)}
        value={value.value}
        label={value.label}
        name={name}
        types={boxTypes}
        onChange={onChildChange}
      />
    </li>
  });

  const props = { checkBoxes };

  return <StyeldComponent { ...componentPrps } { ...props }></StyeldComponent>;
}
export default Container;