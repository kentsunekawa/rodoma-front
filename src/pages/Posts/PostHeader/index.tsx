import React from 'react';
import { useSelector } from 'react-redux';
import Styled from 'styled-components';

import { PostData } from 'types';
import { createCategoryTagList, RELEASE_STATUS } from 'utils';
import { categoryTreeSelector } from 'state/modules/app';

import UserBlock from 'components/blocks/UserBlock';
import TagList from 'components/blocks/TagList';
import Paragraph from 'components/elements/Paragraph';
import CircleButton from 'components/elements/buttons/CircleButton';
import { IconEdit } from 'components/elements/icons';
import { StyleType } from 'components/blocks/TagList/Tag';

import PostEditStatus from '../_id/Edit/PostCreator/PostEditStatus';
import * as styles from './styles';

// component root class name
const CLASSNAME = 'PostHeader';

// declare types

interface ComponentProps {
  isSaved?: boolean;
  isAuthor?: boolean;
  editable?: boolean;
  post: PostData;
  className?: string;
  onTitleClick?: () => void;
}

interface Props extends ComponentProps {
  tabTextList: string[];
  postStatusList: {
    types: StyleType[];
    value: string;
  }[];
}

// dom component
const Component: React.FC<Props> = (props: Props) => (
  <div className={`${CLASSNAME} ${props.className}`}>
    <div className="inner">
      <div className="info" onClick={props.onTitleClick && props.onTitleClick}>
        {props.isAuthor && (
          <CircleButton
            types={['xs', 'gray_dark']}
            link={`/roadmaps/${props.post.id}/edit`}
            className="linkToEdit"
          >
            <IconEdit />
          </CircleButton>
        )}
        <Paragraph className="title" types={['subTitle']}>
          {props.post.title}
        </Paragraph>
        <TagList
          tagTypes={['gradient']}
          types={['alignLeft']}
          values={props.tabTextList}
          // maxLength={7}
        />
      </div>
      <div className="iconArea">
        {props.editable ? (
          <PostEditStatus
            isSaved={props.isSaved ? true : false}
            eyeCatchUrl={props.post.eye_catch_url}
            postStatusList={props.postStatusList}
          />
        ) : (
          <UserBlock
            linkable={!props.editable}
            userId={props.post.user.id}
            userName={props.post.user.name}
            icon_url={props.post.user.icon_url}
            types={['alignCenter', 'm']}
          />
        )}
      </div>
      {props.post.created_at && props.post.updated_at && (
        <div className="date">
          作成日: {props.post.created_at.split(' ')[0]} / 更新日:{' '}
          {props.post.updated_at.split(' ')[0]}
        </div>
      )}
    </div>
  </div>
);

// styled component
const StyeldComponent = Styled(Component)`
  ${styles.base}
`;

// container component
const Container: React.FC<ComponentProps> = (componentProps) => {
  const { post, isSaved } = componentProps;

  const categoryTree = useSelector(categoryTreeSelector);
  const tabTextList = createCategoryTagList(post.category_id, post.specialty_id, categoryTree);

  const postStatusList = (() => {
    return [
      {
        types: isSaved ? ['success'] : ['error'],
        value: isSaved ? '保存済' : '未保存',
      },
      {
        types: post.release_status === 0 ? [] : ['success'],
        value: RELEASE_STATUS[post.release_status],
      },
    ];
  })() as {
    types: StyleType[];
    value: string;
  }[];

  const props = { tabTextList, postStatusList };

  return <StyeldComponent {...componentProps} {...props}></StyeldComponent>;
};
export default Container;
