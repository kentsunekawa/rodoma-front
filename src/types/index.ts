import * as ResponseData from "./ResponseData";
import { RELEASE_STATUS } from "utils";

export type Mode = "light" | "dark";

export type Profile = {
  id: number;
  user_id: number;
  category_id: number;
  specialty_id: number;
  catch_copy: string;
  description: string;
  sns: {
    profile_id: number;
    sns_id: number;
    url: string;
  }[];
};

export interface UserData {
  id: number;
  name: string;
  email: string;
  email_verified_at: string | null;
  icon_url: string;
  profile: Profile;
}

export interface UserData_overview {
  id: number;
  name: string;
  icon_url: string;
}

export type ReleaseStatus = 0 | 1 | 2;

export interface PostData {
  id: number;
  user_id: number;
  category_id: number;
  specialty_id: number;
  release_status: ReleaseStatus;
  title: string;
  description: string;
  eye_catch_url: string;
  created_at: string | null;
  updated_at: string | null;
  likes_count: number | null;
  marks_count: number | null;
  user: UserData_overview;
  subjects: Subject[];
  allowedUsers: UserData_overview[];
}

export interface PostData_overview {
  id: number;
  title: string;
  release_status: keyof typeof RELEASE_STATUS;
  category_id: number;
  specialty_id: number;
  likes_count: number;
  marks_count: number;
  eye_catch_url: string;
  user: UserData_overview;
}

export interface PostData_min {
  id: number;
  title: string;
}

export interface Subject {
  id: number;
  post_id: number;
  label: SubjectLabel;
  linked_post_id: number | null;
  renge_start: number;
  renge_end: number;
  title: string;
  description: string;
}

export interface Specialty {
  id: number;
  name: string;
}

export interface Category {
  id: number;
  name: string;
  specialties: Specialty[];
}

export type CategoryTree = Category[];

export interface Comment {
  id: number;
  comment: string;
  created_at: string;
  user: UserData_overview;
}

export type PostSearchSortKeys = "created_at" | "likes_count" | "marks_count";
export type UserSearchSortKeys = "created_at" | "likes_count";
export type KeywordSearchQuery = {
  keyword: string;
  key: "user" | "post";
};

export type SearchQuery = {
  keyword: KeywordSearchQuery;
  category: number;
  specialty: number;
  orderBy: PostSearchSortKeys | UserSearchSortKeys | "";
};

export type MessageType = "error" | "success";

export interface Message {
  isShow: boolean;
  type: MessageType;
  message: string;
}

export interface SigninInfo {
  email: string;
  password: string;
}

export interface SignupInfo {
  name: string;
  email: string;
  password: string;
  password_confirmation: string;
}

export interface ResetPassInfo {
  token: string;
  email: string;
  password: string;
  password_confirmation: string;
}

export interface CommenInfo {
  user_id: number;
  comment: string;
}

export type Chart = "guntt" | "pie";

export type Size = "xxl" | "xl" | "l" | "m" | "s" | "xs" | "xxs";

export type SubjectLabel = null | 1 | 2 | 3 | 4 | 5 | 6 | 7;

export type Sns = {
  id: number;
  name: string;
};

export interface ValidateStatus<T> {
  isInvalid: boolean;
  errors: T;
}

export interface ValidateErrorStatus {
  suiteName: string;
  validateState: {
    isInvalid: boolean;
    errors: {
      [key: string]: string[];
    } | null;
  };
}

export type BreadCrumbList = {
  id: number;
  name: string;
}[];

export interface RelationData {
  id: number;
  follower_id: number;
  followed_id: number;
}

export { ResponseData };
