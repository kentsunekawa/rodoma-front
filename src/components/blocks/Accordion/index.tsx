import React, { useState, useRef } from 'react';
import Styled from 'styled-components';
import * as styles from './styles';
import TextButton from 'components/elements/buttons/TextButton';
import Paragraph from 'components/elements/Paragraph';
import CircleButton from 'components/elements/buttons/CircleButton';
import { IconUp, IconDown } from 'components/elements/icons';
import { toggleAccordion } from 'components/animations';

// component root class name
const CLASSNAME = 'Accordion';

// declare types
type StyleType = 'nega';

interface ComponentProps {
  isOpen?: boolean;
  parent: {
    id: number;
    name: string;
  };
  childList: {
    id: number;
    name: string;
  }[];
  onClick: (parentId: number, childId?: number) => void;
  className?: string;
  types?: StyleType[];
}

interface Props extends ComponentProps {
  localIsOpen: boolean;
  toggle: (e: React.MouseEvent<HTMLButtonElement>) => void;
  dom: {
    child: React.Ref<HTMLDivElement>;
    inner: React.Ref<HTMLDivElement>;
  };
}

// dom component
const Component: React.FC<Props> = (props: Props) => (
  <div className={`${CLASSNAME} ${props.className}`}>
    <div className="parent">
      <TextButton onClick={() => props.onClick(props.parent.id)}>
        <Paragraph types={props.types}>{props.parent.name}</Paragraph>
      </TextButton>
      <CircleButton onClick={props.toggle} types={['outline', 's']}>
        {props.localIsOpen ? <IconUp /> : <IconDown />}
      </CircleButton>
    </div>

    <div className={`child${props.localIsOpen ? ' -open' : ''}`} ref={props.dom.child}>
      <div className="inner" ref={props.dom.inner}>
        <ul className="list">
          {props.childList.map((child, i) => {
            return (
              <li className="item" key={i}>
                <TextButton onClick={() => props.onClick(props.parent.id, child.id)}>
                  <Paragraph types={props.types}>{child.name}</Paragraph>
                </TextButton>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  </div>
);

// styled component
const StyeldComponent = Styled(Component)`
  ${styles.base}
`;

// container component
const Container: React.FC<ComponentProps> = (componentProps) => {
  const { isOpen } = componentProps;

  const [localIsOpen, setLocalIsOpen] = useState(isOpen ? true : false);

  const dom = {
    child: useRef<HTMLDivElement>(null),
    inner: useRef<HTMLDivElement>(null),
  };

  const toggle = () => {
    if (dom.child.current && dom.inner.current) {
      toggleAccordion(dom.child.current, dom.inner.current, !localIsOpen);
    }
    setLocalIsOpen(localIsOpen ? false : true);
  };

  const props = { localIsOpen, toggle, dom };

  return <StyeldComponent {...componentProps} {...props} />;
};
export default Container;
