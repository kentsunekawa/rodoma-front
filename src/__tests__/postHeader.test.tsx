import React from 'react';
import { render, fireEvent, cleanup, getByTestId } from 'components/TestWrapper';
import { PostData } from 'types';
import { Component } from 'components/blocks/PostHeader';

const mockPostData: PostData = {
  id: 1,
  user_id: 1,
  category_id: 1,
  specialty_id: 1,
  release_status: 0,
  title: 'post title',
  description: 'post description',
  eye_catch_url: 'http://img.rodoma.net/testtest',
  created_at: '2021-01-01',
  updated_at: '2021-01-01',
  likes_count: 0,
  marks_count: 0,
  user: {
    id: 1,
    name: 'test user',
    icon_url: '',
  },
  subjects: [],
  allowedUsers: [],
};

const mockBaseProps = {
  post: mockPostData,
  tabTextList: ['category01', 'specialty01'],
  postStatusList: [
    {
      types: [],
      value: 'status01',
    },
  ],
};

afterEach(cleanup);

describe('PostHeader', () => {
  test('snapshot test', () => {
    const tree = render(<Component {...mockBaseProps} />);
    expect(tree).toMatchSnapshot();
  });

  test('event was called', () => {
    const props = { ...mockBaseProps, onTitleClick: jest.fn() };
    const tree = render(<Component {...props} />);
    fireEvent.click(tree.getByTestId('PostHeader__title'));
    expect(props.onTitleClick.mock.calls.length).toBe(1);
  });

  test('status check saved unreleased', () => {
    const tree = render(
      <Component
        {...{
          ...mockBaseProps,
          isSaved: true,
          editable: true,
          postStatusList: [
            {
              types: ['success'],
              value: '保存済',
            },
            {
              types: [],
              value: '未公開',
            },
          ],
        }}
      />
    );

    const tags = tree.container.querySelectorAll('.PostEditStatus .Tag');

    const tag01 = tags[0];
    expect(tag01.querySelector('.text')).toHaveTextContent('保存済');

    const tag02 = tags[1];
    expect(tag02.querySelector('.text')).toHaveTextContent('未公開');
  });

  test('status check when author', () => {
    const tree = render(
      <Component
        {...{
          ...mockBaseProps,
          isAuthor: true,
        }}
      />
    );

    const editButton = tree.container.querySelector('.linkToEdit') as HTMLButtonElement;
    expect(tree.container).toContainElement(editButton);
  });
});
