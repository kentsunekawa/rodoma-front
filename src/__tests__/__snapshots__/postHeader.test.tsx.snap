// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostHeader snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="PostHeader undefined"
      >
        <div
          class="inner"
        >
          <div
            class="info"
            data-testid="PostHeader__title"
          >
            <p
              class="Paragraph sc-jRQAMF uRnSJ title"
            >
              post title
            </p>
            <div
              class="TagList sc-eCImvq fNyUIZ"
            >
              <ul
                class="list"
              >
                <li
                  class="item"
                >
                  <div
                    class="Tag sc-hKwCoD dloDGW"
                  >
                    <div
                      class="inner"
                    >
                      <span
                        class="text"
                      >
                        category01
                      </span>
                    </div>
                  </div>
                </li>
                <li
                  class="item"
                >
                  <div
                    class="Tag sc-hKwCoD dloDGW"
                  >
                    <div
                      class="inner"
                    >
                      <span
                        class="text"
                      >
                        specialty01
                      </span>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="iconArea"
          >
            <div
              class="UserBlock sc-dkPtyc jpLAvs"
            >
              <div
                class="icon"
              >
                <a
                  href="/users/1"
                >
                  <div
                    class="UserIcon sc-bdvvaa clKduF userIcon"
                  >
                    <div
                      class="inner"
                    />
                  </div>
                </a>
              </div>
              <p
                class="name"
              >
                test user
              </p>
            </div>
          </div>
          <div
            class="date"
          >
            作成日: 
            2021-01-01
             / 更新日:
             
            2021-01-01
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="PostHeader undefined"
    >
      <div
        class="inner"
      >
        <div
          class="info"
          data-testid="PostHeader__title"
        >
          <p
            class="Paragraph sc-jRQAMF uRnSJ title"
          >
            post title
          </p>
          <div
            class="TagList sc-eCImvq fNyUIZ"
          >
            <ul
              class="list"
            >
              <li
                class="item"
              >
                <div
                  class="Tag sc-hKwCoD dloDGW"
                >
                  <div
                    class="inner"
                  >
                    <span
                      class="text"
                    >
                      category01
                    </span>
                  </div>
                </div>
              </li>
              <li
                class="item"
              >
                <div
                  class="Tag sc-hKwCoD dloDGW"
                >
                  <div
                    class="inner"
                  >
                    <span
                      class="text"
                    >
                      specialty01
                    </span>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="iconArea"
        >
          <div
            class="UserBlock sc-dkPtyc jpLAvs"
          >
            <div
              class="icon"
            >
              <a
                href="/users/1"
              >
                <div
                  class="UserIcon sc-bdvvaa clKduF userIcon"
                >
                  <div
                    class="inner"
                  />
                </div>
              </a>
            </div>
            <p
              class="name"
            >
              test user
            </p>
          </div>
        </div>
        <div
          class="date"
        >
          作成日: 
          2021-01-01
           / 更新日:
           
          2021-01-01
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
